--- a/drivers/xenon/smc-core.c
+++ b/drivers/xenon/smc-core.c
@@ -106,12 +106,11 @@
 }
 
 static unsigned char * _xenon_smc_cache_lookup(unsigned char *msg) {
-	unsigned char *ptr = smc_reply[0];
-
-	while (ptr[0]) {
-		if (msg[0] == ptr[0])
-			return ptr;
-		ptr += 16;
+	unsigned int i;
+
+	for (i = 0; i < ARRAY_SIZE(smc_reply); i++) {
+		if (msg[0] == smc_reply[i][0])
+			return smc_reply[i];
 	}
 	return NULL;
 }
